File: 42-trapping-rain-water/solution2.cpp
// Runtime: 8ms, Beats 80.51% of users with C++
// Memory: 22.40MB, Beats 8.92% of users with C++

class Solution {
public:
    int trap(vector<int>& height) {
        int water=0;
        int p=0,q=height.size()-1;
        int maxLeft=height[p],maxRight=height[q];
        while (p<q)
        {
            if(maxLeft<=maxRight)
            {
                p++;
                maxLeft=max(maxLeft,height[p]);
                water+=maxLeft-height[p];
                
            }
            else
            {
                q--;
                maxRight=max(maxRight,height[q]);
                water+=maxRight-height[q];
            }
        }
        return water;
    }
};

File: 2540-minimum-common-value/solution2.py
# Runtime: 335ms, Beats 94.06% of users with Python3
# Memory: 47.04MB, Beats 5.87% of users with Python3

class Solution:
    def getCommon(self, nums1: List[int], nums2: List[int]) -> int:
        s=set(nums1)-set(nums2)
        for i in range(len(nums1)):
            if nums1[i] not in s:
                return nums1[i]
        return -1